### PR TITLE
Switch to using raw HttpClientHandler on all platforms

### LINKED ISSUE #959: Custom validation of server SSL Certificate not working.
Support for this needs to be added to HttpClient API.
Issue opened in corefx repo: dotnet/corefx#4476

Failing test case...
-   Security.TransportSecurity.Tests#HttpsTests.ServerCertificateValidation_EchoString

### PR DESCRIPTION
**Pull request by mconnew**
The HttpClientHandler on UWP was updated to handle client certificates and supports setting a server certificate validation callback.
Fixes #959

### CODE REVIEW

File: src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpsChannelFactory.cs [hunk 1]
```diff
@@ -227,30 +227,31 @@ internal override ServiceModelHttpMessageHandler GetHttpMessageHandler(EndpointA
...
-                throw ExceptionHelper.PlatformNotSupported("Directly setting the Certificate is not supported for HTTP yet. Use X509CertificateInitiatorClientCredential.SetCertificate instead");
-            }
+                Thumbprint = certificate.GetCertHash(),
+                IncludeDuplicates = false,
+                StoreName = "MY"
+            };
+
+            if (CertificateStores.FindAllAsync(query).AsTask().GetAwaiter().GetResult().Count == 0)
+            {
+                throw ExceptionHelper.PlatformNotSupported("Certificate could not be found in the MY store.");
```
Comment (shmao) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpsChannelFactory.cs]: Is the exception still PNSE?

**Comment by shmao** [comment]
@mconnew can you please confirm if the tests passed when targeting uapaot with your fix?

File: src/System.Private.ServiceModel/src/System/ServiceModel/Security/X509CertificateInitiatorClientCredential.cs [hunk 2]
```diff
@@ -27,6 +27,14 @@ internal X509CertificateInitiatorClientCredential(X509CertificateInitiatorClient
             _isReadOnly = other._isReadOnly;
         }
 
+        internal bool CloneCertificate
+        {
+            get
+            {
+                return RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+            }
+        }
```
Comment (zhenlan) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Security/X509CertificateInitiatorClientCredential.cs]: Probably worth adding some comments to explain why this should return true if it's windows.

File: src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpsChannelFactory.cs (hunk 1 above)
Comment (mconnew) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpsChannelFactory.cs]: Yes, we don't support this on this platform.

File: src/System.Private.ServiceModel/src/System/ServiceModel/Security/X509CertificateInitiatorClientCredential.cs (hunk 2 above)
Comment (mconnew) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Security/X509CertificateInitiatorClientCredential.cs]: Will do.

**Pull request merged**